src/webgl-surface/buffers/static/uniform-ribbon-buffer.ts

```diff
@@ -0,0 +1,251 @@
+import { flatten } from 'ramda';
+import { IUniform, Mesh, ShaderMaterial, TriangleStripDrawMode, Vector4 } from 'three';
+import { ReferenceColor } from '../../drawing/reference/reference-color';
+import { RibbonShape } from '../../drawing/shape/ribbon-shape';
+import { AtlasManager } from '../../drawing/texture/atlas-manager';
+import { AttributeSize, BufferUtil, UniformAttributeSize } from '../../util/buffer-util';
+import { BaseBuffer } from '../base-buffer';
+
+const debug = require('debug')('webgl:uniform-ribbon-buffer');
+
+const MAX_SEGMENTS_PER_CURVE = 100;
+const VERTICES_PER_SEGMENT = 6;
+const VERTICES_PER_CURVE = VERTICES_PER_SEGMENT * MAX_SEGMENTS_PER_CURVE;
+
+function isCluster(val: RibbonShape<any>[] | RibbonShape<any>[][]): val is RibbonShape<any>[][] {
+  return Array.isArray(val[0]);
+}
+
+export class UniformRibbonBuffer extends BaseBuffer <RibbonShape<any> | RibbonShape<any>[], Mesh> {
+
+  init(material: ShaderMaterial, unitCount: number, shared?: UniformRibbonBuffer) {
+    this.bufferItems = BufferUtil.makeBufferItems();
+    debug('init ribbon~');
+    // Declare the structure of the uniform data in the instanceData uniform
```

Comment:
This is where a newline would be nice, before a large scoped block. Also, you may consider deleting the debug.